weblate/checks/chars.py

```diff
@@ -290,6 +294,21 @@ def check_single(self, source, target, unit):
         return self.check_chars(source, target, -1, ("!", "！", "՜", "᥄", "႟", "߹"))
 
 
+class EndInterrobangCheck(TargetCheck):
+    """Check for final interrobang expression."""
+
+    check_id = "end_Interrobang"
+    name = gettext_lazy("Mismatched interrobang")
+    description = gettext_lazy(
+        "Source and translation do not both end with an interrobang expression"
+    )
+
+    def check_single(self, source, target, unit):
+        if not source or not target:
+            return False
+        return self.check_interrobang(source, target)
```

Comment:
Move it here (and simplify):

```suggestion
        strings = ("!?", "?!")
        return (source[-2:] in strings) != (target[-2:] in strings)
```

Reply:
Adding support for pre-combined ⁈ (U+2048) and ⁉ (U+2049) might be reasonable as well (see https://github.com/WeblateOrg/weblate/issues/10296).